vnext/Microsoft.ReactNative/ABIViewManager.cpp

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "ABIViewManager.h"
+
+#include <ReactUWP/Utils/ValueUtils.h>
+#include "ReactSupport.h"
+
+namespace winrt::Microsoft::ReactNative::Bridge {
+
+ABIViewManager::ABIViewManager(
+    const std::shared_ptr<react::uwp::IReactInstance> &reactInstance,
+    const winrt::Microsoft::ReactNative::Bridge::IViewManager &viewManager)
+    : Super(reactInstance) {
+  if (nullptr == viewManager) {
+    throw winrt::hresult_null_argument(L"viewManager");
+  }
+  m_viewManager = viewManager;
+  m_name = to_string(m_viewManager.Name());
+  m_nativeProps = m_viewManager.NativeProps();
+}
+
+const char *ABIViewManager::GetName() const {
+  return m_name.c_str();
+}
+
+folly::dynamic ABIViewManager::GetExportedViewConstants() const {
+  folly::dynamic parent = Super::GetExportedViewConstants();
+
+  auto outerChild = m_viewManager.ExportedViewConstants();
+  for (auto const &pair : outerChild) {
+    std::string key = to_string(pair.Key());
```

Comment:
>    std::string [](start = 0, length = 15)

const auto&

Reply:
No, keeping a temporary in a reference will cause crashes. std::string is much better and more informative.

---
In reply to: [337875961](https://github.com/microsoft/react-native-windows/pull/3490#discussion_r337875961) [](ancestors = 337875961)